Sources/Quick/Configuration/QuickConfiguration.swift

```diff
@@ -25,10 +25,11 @@ internal func qck_enumerateSubclasses<T: AnyObject>(_ klass: T.Type, block: (T.T
     var subclass, superclass: AnyClass!
     for i in 0..<classesCount {
         subclass = classes[Int(i)]
-        superclass = class_getSuperclass(subclass)
-        if superclass === klass {
-            block(subclass as! T.Type) // swiftlint:disable:this force_cast
-        }
+
+        guard let superclass = class_getSuperclass(subclass),
+            superclass === klass else { return }
+
+        block(subclass as! T.Type) // swiftlint:disable:this force_cast
```

Comment:
I'm not sure I understand how this fixes the problem.

So, here's the code before:

```swift
// 1. Get the superclass.
superclass = class_getSuperclass(subclass)
// 2. If the superclass is the same as the class passed into this function...
if superclass === klass {
    // 3. ...then force-cast the subclass.
    block(subclass as! T.Type)
}
```

And here's the new version:

```swift
// 1. Get the superclass and check if it's the same as the class passed into
//    this function. If it isn't, exit early.
guard let superclass = class_getSuperclass(subclass),
    superclass === klass else { return }
// 2. Otherwise, the superclass is the same as the class passed into this function...
// 3. ...then force-cast the subclass.
block(subclass as! T.Type)
```

Reading the two, it seems like they basically do the same thing...?

However, I think maybe the difference is that you `return` instead of `continue`. So I think the actual assertion failure happens later in the iteration. In other words, maybe your change prevents the error because it prevents this loop from going through all of the classes, and therefore it never reaches the class that causes the error.

Or am I misreading this? Let me know, thanks!